tests/integration/abi_router_test.py

```diff
@@ -0,0 +1,466 @@
+from dataclasses import dataclass
```

Comment:
New test-file broken off from `tests/integration/abi_router_test.py` but with the same tests, only refactored to use `class Simulation` and `class ABIMethodCallStrategy`